src/pymovements/events/events.py

```diff
@@ -88,6 +88,7 @@ def __init__(
 
             # Make sure lengths of onsets and offsets are equal.
             if onsets is not None:
+                assert offsets is not None
```

Comment:
Please use `if offsets is None: raise ValueError()` here.

Asserts in production code are problematic and should be used exclusively for stuff that should never happen independent of user input.
If something is failing due to user input error or other dependent systems please use an exception instead.

See this article here: https://www.inspiredpython.com/article/assertions-and-exceptions-are-not-the-same

Reply:
this happens in [line 86-87](https://github.com/aeye-lab/pymovements/pull/295/files#diff-82cc6c96b216cd7ff8cefcac7928eb28601370e2180b3729e129a73e6c31fe7fR86-R87)

```python
# Make sure that if either onsets or offsets is None, the other one is None too.
checks.check_is_none_is_mutual(onsets=onsets, offsets=offsets)
```

Reply:
I'll pushed another commit with a comment explicitly pointing at line 87 that it is within mypy and in line 91 there can not be any none offsets

Reply:
ah sorry I didn't see that correctly, so everything's fine and I'll merge after the tests complete

Thanks a lot again!